web/src/app/api/[...path]/route.ts

```diff
@@ -0,0 +1,133 @@
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function PUT(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function PATCH(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function DELETE(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function HEAD(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+export async function OPTIONS(
+  request: NextRequest,
+  { params }: { params: { path: string[] } }
+) {
+  return handleRequest(request, params.path);
+}
+
+async function handleRequest(request: NextRequest, path: string[]) {
+  if (process.env.NODE_ENV !== "development") {
+    return NextResponse.json(
+      {
+        message:
+          "This API is only available in development mode. In production, something else (e.g. nginx) should handle this.",
+      },
+      { status: 404 }
+    );
+  }
+
+  try {
+    const backendUrl = new URL(`${INTERNAL_URL}/${path.join("/")}`);
+
+    // Get the URL parameters from the request
+    const urlParams = new URLSearchParams(request.url.split("?")[1]);
+
+    // Append the URL parameters to the backend URL
+    urlParams.forEach((value, key) => {
+      backendUrl.searchParams.append(key, value);
+    });
+
+    const response = await fetch(backendUrl, {
+      method: request.method,
+      headers: request.headers,
+      body: request.body,
+      // @ts-ignore
+      duplex: "half",
+    });
+
+    // Check if the response is a stream
+    if (
+      response.headers.get("Transfer-Encoding") === "chunked" ||
+      response.headers.get("Content-Type")?.includes("stream")
+    ) {
+      // If it's a stream, create a TransformStream to pass the data through
+      const { readable, writable } = new TransformStream();
+      response.body?.pipeTo(writable);
+
+      return new NextResponse(readable, {
+        status: response.status,
+        headers: response.headers,
+      });
+    } else {
+      // If it's not a stream, handle it based on content type
+      const contentType =
+        response.headers.get("content-type") || "application/json";
+
+      if (contentType.startsWith("image/")) {
+        // For image types, return the response as-is
+        return new NextResponse(response.body, {
+          status: response.status,
+          headers: response.headers,
+        });
+      } else {
+        // For other types, handle as before
+        const data = await response.text();
+
+        return new NextResponse(data, {
+          status: response.status,
+          headers: {
+            "Content-Type": contentType,
+          },
```

Comment:
Think we're not handling all the data types (e.g. 204 response code, json content type) - maybe we could have a stricter check for using response.text() and default to response.body more often?

Reply:
I believe those should all be handled as expected (e.g. all our APIs return JSONs, but it all works as expected + 204 is "no content" which is handled fine here as well). Afaik most response types are just text / can be represented fully as just text

Reply:
using body by default now